### PR TITLE
Fix for the latest ansible-core version

### PR DESCRIPTION
**Pull request by KosukeOkamoto**
## Related Issue
Supports #ISSUE_NUMBER

## Related PRs
This PR is not dependent on any other PR

## What this PR does / why we need it

ansible watch manager does not work with ansible-core >= 2.15.7. This PR address some breaking changes introduced into the recent ansible-core versions.
- Removed problematic brackets for [the breaking change that does not allow nested templating](https://github.com/ansible/ansible/blob/v2.15.7/changelogs/CHANGELOG-v2.15.rst#breaking-changes--porting-guide)
- Correct shebang to directly point system python as mentioned in [the ansible 2.17 issue](https://github.com/ansible/ansible/issues/83603).

## Special notes for your reviewer
I patched the code and confirmed ansible watch manager works with the latest ansible-core in my project.

## If applicable**
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility
    - Works with ansible-core 2.15.6

## What gif most accurately describes how I feel towards this PR?
Example of a gif

### CODE REVIEW

File: oper8/watch_manager/ansible_watch_manager/resources/roles/oper8_app/tasks/main.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@
   # 1) Define the 'develop_shell' ansible variable in the playbook invocation
   # 2) Define the per-playbook 'enable_ansible_vcs' variable
   # 3) Set the ANSIBLE_VCS environment variable
-  when: develop_shell is not defined and "{{ enable_ansible_vcs | default('true', True) }}" != "false" and "{{ lookup('env', 'ANSIBLE_VCS') | default('true', True) }}" != "false"
+  when: develop_shell is not defined and "enable_ansible_vcs | default('true', True)" != "false" and "lookup('env', 'ANSIBLE_VCS') | default('true', True)" != "false"
```
Comment (HonakerM) [inline oper8/watch_manager/ansible_watch_manager/resources/roles/oper8_app/tasks/main.yml:15]: Have we checked that these functions are still called inside quotes? Like I'm a little worried that ansible is just doing a direct string comparison between `"enable_ansible_vcs | default('true', True)"` and `"false"`

File: oper8/watch_manager/ansible_watch_manager/modules/k8s_application.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
```
Comment (gabe-l-hart) [inline oper8/watch_manager/ansible_watch_manager/modules/k8s_application.py:1]: I'm curious why this change was added. Does the new `ansible` base image not have `/usr/bin/env` in it? In general, that's the preferred method for a shebang pointing at the generic version of `python` or `bash` in a script like this.

File: oper8/watch_manager/ansible_watch_manager/resources/roles/oper8_app/tasks/main.yml (hunk 1 above)
Comment (gabe-l-hart) [reply oper8/watch_manager/ansible_watch_manager/resources/roles/oper8_app/tasks/main.yml:15]: Hm, yeah, we definitely need to check this one too.

File: oper8/watch_manager/ansible_watch_manager/modules/k8s_application.py (hunk 2 above)
Comment (KosukeOkamoto) [reply oper8/watch_manager/ansible_watch_manager/modules/k8s_application.py:1]: The image has `/usr/bin/env` and it works if I run manually in the container. I also think it's usually right way. However, ansible raise error as `/usr/bin/env python` not found. It is currently not allowed in ansible according to [this issue comment](https://github.com/ansible/ansible/issues/83603).

**Review by gabe-l-hart — CHANGES_REQUESTED** [review]
Thanks for the context on the shebang! I had no idea `ansible` was this opinionated about the version of python. One quick suggestion to add the coding and a note to the docs.

**Review by gabe-l-hart — APPROVED** [review]
Looks good!

**Comment by gabe-l-hart** [comment]
@KosukeOkamoto Looks like we need to add a `ruff` linting rule to fix [the linter](https://github.com/IBM/oper8/actions/runs/11827115345/job/32994959426?pr=139). I think it should go [here](https://github.com/IBM/oper8/blob/main/pyproject.toml#L151) to ignore rule `UP009` for that file specifically (with a comment about this being required by Ansible).

**Comment by KosukeOkamoto** [comment]
@gabe-l-hart Thank you for your detailed suggestion! I made the changes and the ruff formatting and linting passed in local.

**Review by gabe-l-hart — APPROVED** [review]
